beta/nncf/tensorflow/sparsity/rb/algorithm.py

```diff
@@ -0,0 +1,159 @@
+    def __init__(self, config):
+        if isinstance(tf.distribute.get_strategy(), tf.distribute.MirroredStrategy):
+            raise Exception('RB sparsity algorithm do not support the distributed mode with mirrored strategy')
+        super().__init__(config)
+        self.ignored_scopes = self.config.get('ignored_scopes', [])
+
+    def get_transformation_layout(self, model):
+        nxmodel = convert_keras_model_to_nxmodel(model)
+        transformations = TransformationLayout()
+        shared_nodes = set()
+
+        for node_name, node in nxmodel.nodes.items():
+            original_node_name, _ = get_original_name_and_instance_index(node_name)
+            if (node['type'] not in SPARSITY_LAYERS or
+                is_ignored(node_name, self.ignored_scopes) or
+                original_node_name in shared_nodes):
+                continue
+
+            if node['is_shared']:
+                shared_nodes.add(original_node_name)
+
+            weight_attr_name = SPARSITY_LAYERS[node['type']]['weight_attr_name']
+            name = node_name + OP_NAME
+            assert name not in self.op_names, 'duplicate names of RBSparsifyingWeight'
+            self.op_names.add(name)
+
+            transformations.register(
+                TFInsertionCommand(
+                    target_point=TFLayerWeight(original_node_name, weight_attr_name),
+                    callable_object=RBSparsifyingWeight(name),
+                    priority=TransformationPriority.SPARSIFICATION_PRIORITY
+                ))
+
+        return transformations
+
+    def build_controller(self, model) -> BaseSparsityController:
+        """
+        Should be called once the compressed model target_model is fully constructed
+        """
+
+        return RBSparsityController(model, self.config, self.op_names)
+
+
+class RBSparsityController(BaseSparsityController):
+    def __init__(self, target_model, config, op_names: set):
+        super().__init__(target_model, op_names)
+        params = config.get('params', {})
+        self.sparsity_init = config.get('sparsity_init', 0)
+        sparsity_level_mode = params.get('sparsity_level_setting_mode', 'global')
+
+        if sparsity_level_mode == 'local':
+            raise NotImplementedError
+
+        target_ops = apply_fn_to_op_weights(target_model, op_names, lambda x: (x['mask'], x['trainable']))
+        self._loss = SparseLoss(target_ops)
+        schedule_type = params.get('schedule', 'exponential')
+        scheduler_cls = SPARSITY_SCHEDULERS.get(schedule_type)
+        self._scheduler = scheduler_cls(self, params)
+        self.set_sparsity_level(self.sparsity_init)
+
+    def set_sparsity_level(self, sparsity_level):
+        self._loss.set_target_sparsity_loss(sparsity_level)
+
+    def freeze(self):
+        self._loss.disable()
+
+    def statistics(self, quickly_collected_only: bool = False):
+        return super().statistics(quickly_collected_only=quickly_collected_only)
```

Comment:
Seem like this method is not needed here because it uses implementation from the superclass only.

Reply:
Removed it